File: Unidad 2-Estructuras de Control/c/Ej15.c
/*
Autor: Guillermo Canto 2/3/2019
Entradas: 2 fechas representadas por cantidades enteras con el formato dd/mm/aaaa
Salidas: La fecha mayor, o si ambas son iguales
Procedimiento general: Compara los años, meses y dias para determinar si son iguales, de no serlo 
determina la fecha mayor.
*/

#include <stdio.h>

int main(int argc, char *argv[]) {
	int day1;
	int day2;
	int month1;
	int month2;
	int year1;
	int year2;
	printf("Ingrese las fechas con el formato dd/mm/aaaa.\n");
	do {
		printf("Ingrese el dia 1: ");
		scanf("%d", &day1);
	} while ( day1 <= 0 || day1 >31 );
	do {
		printf("Ingrese el mes 1: ");
		scanf("%d", &month1);
	}  while ( month1 <= 0 || month1 > 12);
	do {
		printf("Ingrese el año 1: ");
		scanf("%d", &year1);
	} while ( year1 < 0 );
	do {
		printf("Ingrese el dia 2: ");
		scanf("%d", &day2);
	} while ( day2 <= 0 || day2 >31 );
	do {
		printf("Ingrese el mes 2: ");
		scanf("%d", &month2);
	}  while ( month2 <= 0 || month2 > 12);
	do {
		printf("Ingrese el año 2: ");
		scanf("%d", &year2);
	} while ( year2 < 0 );
	

	if (year1 != year2){
		if (year1 > year2){
			printf("La fecha mayor es: %d / %d / %d ",day1,month1,year1 );
		} else{
			printf("La fecha mayor es: %d / %d / %d ",day2,month2,year2 );
		}
	} else if (month1 != month2){
		if (month1 > month2){
			printf("La fecha mayor es: %d / %d / %d ",day1,month1,year1 );
		} else {
			printf("La fecha mayor es: %d / %d / %d ",day2,month2,year2 );
		}
	} else if (day1 !=  day2){
		if (day1 > day2){
			printf("La fecha mayor es: %d / %d / %d ",day1,month1,year1 );
		} else {
			printf("La fecha mayor es: %d / %d / %d ",day2,month2,year2 );
		}
	} else {
		printf("Las fechas son iguales"); 
	}
	return 0;
}

File: Unidad 2-Estructuras de Control/lua/Ej15.lua
--[[
Autor: Guillermo Canto 2/3/2019
Entradas: 2 fechas representadas por cantidades enteras con el formato dd/mm/aaaa
Salidas: La fecha mayor, o si ambas son iguales
Procedimiento general: Compara los años, meses y dias para determinar si son iguales, de no serlo 
determina la fecha mayor.
]]
--Inputs
day1 = io.read("*n")
month1 = io.read("*n")
year1 = io.read("*n")
day2 = io.read("*n")
month2 = io.read("*n")
year2 = io.read("*n")
--Process
if year1 ~= year2 then
  if year1 > year2 then
    io.write("La fecha mayor es " .. day1 .. "/" .. month1 .. "/" .. year1 )
  else
    io.write("La fecha mayor es " .. day2 .. "/" .. month2 .. "/" .. year2 )
  end
elseif month1 ~= month2 then
  if month1 > month2 then
    io.write("La fecha mayor es " .. day1 .. "/" .. month1 .. "/" .. year1 )
  else
    io.write("La fecha mayor es " .. day2 .. "/" .. month2 .. "/" .. year2 )
  end
elseif day1 ~= day2 then
  if day1 > day2 then
  io.write("La fecha mayor es " .. day1 .. "/" .. month1 .. "/" .. year1 )
  else
    io.write("La fecha mayor es " .. day2 .. "/" .. month2 .. "/" .. year2 )
  end
else 
  io.write("Las fechas son iguales")
end